react/src/utils/object-utils.ts

```diff
@@ -0,0 +1,19 @@
+export function isDeepEqual(obj1: any, obj2: any): boolean {
+    if (obj1 === obj2) return true
+
+    if (typeof obj1 !== 'object' || obj1 === null || typeof obj2 !== 'object' || obj2 === null) {
+        return false
+    }
+
+    const keys1 = Object.keys(obj1)
+    const keys2 = Object.keys(obj2)
+
+    if (keys1.length !== keys2.length) return false
```

Comment:
```suggestion
    if (keys1.length !== keys2.length) {
      return false
    }
```
We generally prefer this style in the codebase. (I'm surprised prettier didn't change this!)